### PR TITLE
Implementing 'Webhook Intercepting Webhooks'

### LINKED ISSUE #101794: Add feature to configure a set of webhooks to intercept other webhooks
#### What would you like to be added:
Platform providers should be able to configure a select set of webhooks to intercept other webhoooks as well.

#### Why is this needed:
Currently webhooks [do not intercept other webhook types](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go#L222). This is rightly designed to avoid cycles in the admission flow, and also prevent losing control of a webhook in case of a bad configuration (e.g. a webhook rejects all webhook UPDATEs and DELETEs).

However, it is sometimes necessary to intercept webhooks to enforce policy. For instance, consider a platform validation webhook which only allows a certain set of root users to create pods in kube-system. Now, if a user creates a mutating webhook X which intercepts all pods, it could mutate a root users's pod request and use its privilege to create an arbitrary pod in the restricted kube-system namespace. 

In this case, a platform webhook which intercepts other webhooks would intercept X, and decides to either modify or reject it, so that the enforced policies won't be tampered with.

### PR DESCRIPTION
**Pull request by soorena776**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Platform providers should be able to configure a select set of webhooks to intercept other webhoooks as well.

#### Which issue(s) this PR fixes:
Fixes #101794

#### Does this PR introduce a user-facing change?

```release-note
Users can now optionally specify a set of webhooks names in the AdmissionConfiguration for mutating and/or validating webhook plugins, which could also intercept other webhooks.
```

### CODE REVIEW

**Comment by soorena776** [comment]
I signed it

**Comment by lavalamp** [comment]
Sorry I haven't had a chance to review this yet. I've added it to the agenda backlog for api machinery SIG meeting. I'm not sure if we'll get to it tomorrow or not.

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/types.go [hunk 1]
```diff
@@ -18,12 +18,58 @@ package webhookadmission
...
+	Identifiers []WebhookIdentifier 
+	// Maitainers consist of identities which maintain (i.e. could
+	// add/delete/modify) the webhook intercepting webhooks (WIWs). Normally
+	// webhook admission requests are intercepted by WIWs, and could potentially
+	// be denied (for instance when the webhook is misconfigured). However,
+	// webhook admission requests made by maintainers will *not* be intercepted
+	// by WIWs, enabling them to maintain these webhooks. A request is
+	// considered to be made by a maintainer if either its username is in Users,
+	// or one of its groups is in Groups.
+	Maitainers WebhookMaintainers
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/types.go:53]:
```suggestion
	Maintainers WebhookMaintainers
```

Comment (soorena776) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/types.go]: thanks!

**Comment by soorena776** [comment]
Any chance I could get some reviews on this? :)

**Comment by liggitt** [comment]
/unassign
/assign @dbsmith

for context and api-machinery review

**Comment by lavalamp** [comment]
This fell off my radar, I'll sync up with @soorena776 to figure out what is next.

**Comment by liggitt** [comment]
closing due to inactivity. if this direction is accepted, feel free to reopen.

**Pull request closed without merging**